### ISSUE TITLE
Retrieval-side de-censoring: adaptive retrieval floor / confidence-weighted retrieval for compressed manifolds

### ISSUE THREAD

**Issue by samscarrow** [git-scarrow/fast-associative-memory#80]
## Context
Spun out from #73. The calibration comparison there ([result](https://github.com/git-scarrow/fast-associative-memory/issues/73#issuecomment-4559126243)) showed that `RelativeVigilance` — a **write-gate** policy — does **not** resolve the epoch-0 left-censoring of the blend onset on real embeddings. Margin and relative policies produce a byte-identical blend onset (ρ=0.233 @ epoch 0) on real 20NG + all-MiniLM-L6-v2.

## Why this is a separate problem
Blend onset is a **retrieval-side** phenomenon: off-class softmax vote mass is a function of `inference_temp` and the **resting geometry of the embedding manifold**, which already places off-class neighbors inside the retrieval neighborhood at contraction=0 (real Δ≈0.42, off-class ρ≈0.23 before any drift). Changing which writes are admitted/merged cannot retroactively separate retrieval neighborhoods the encoder collapsed. The write gate is the wrong lever for this failure mode.

## Goal
Reduce / de-censor the retrieval-time blend regime on compressed real manifolds, where the write-gate anchor (#73) provably cannot.

## Candidate approaches to evaluate
- **Adaptive retrieval floor** — extend the live-Δ `RetrievalFloorPolicy` (#74) to set `inference_sim_floor` from the live off-class distribution, so low-margin votes are suppressed rather than blended.
- **Confidence-weighted retrieval** — gate or down-weight the output vote by a calibrated confidence signal (e.g. top-1 vs top-2 margin, or off-class vote-mass), abstaining / sharpening when the neighborhood is ambiguous.
- **Temperature adaptation** — make `inference_temp` manifold-relative instead of the fixed 0.05, so the softmax sharpness tracks the achievable separation.
- **Manifold-relative chimera criterion** — replace the absolute `v_ceiling=0.95` (unreachable on real, max ρ≈0.45) with a relative criterion, so the chimera onset is detectable at all on real manifolds.

## Evaluation
Reuse `benchmarks/probe_contraction.py` (after the #73 instrumentation PR lands occupancy/merge_rate/accuracy columns). Compare against the margin + relative baselines on:
- first blend epoch / observed blend ρ — **is real blend onset still left-censored at epoch 0?** (the headline question the write gate could not move)
- off-class vote mass, frac_blended
- held-out accuracy / label purity under drift
- abstention rate (if a confidence gate is added)

## Scope
Retrieval path only. Write-gate calibration is handled in #73. The two compose: keep `RelativeVigilance` for write hygiene, add a retrieval-side mechanism here for the blend regime.

**Comment by samscarrow** [comment]
## Live-Δ retrieval floor de-censors blend onset on real-20NG

**Headline (positive):** the shipped `RetrievalFloorPolicy` (live-Δ, #74) **does** move the blend onset off epoch 0 on the real 20NG + all-MiniLM-L6-v2 manifold — the failure mode #73 showed the write gate could not fix. Static-floor blend onset remains left-censored at epoch 0; live-Δ floor pushes onset to epoch 4 (margin) or epoch 6 (relative).

### Setup
Ran `benchmarks/probe_contraction.py` 2×2 matrix on real-20NG, matching the #73 calibration setup (seed 0, 30 epochs, `--contraction-end 0.9 --margin-guard 0.05 --percentile 0.95`, default `--k-logit-steps 3.0`, default `inference_temp 0.05`):

```bash
python benchmarks/probe_contraction.py --real \
    --vigilance-policy {margin,relative} \
    --retrieval-floor-policy {static,live-delta} \
    --epochs 30 --contraction-end 0.9 --seed 0
```

### Headline: blend-onset epoch (held-out off-class vote mass ≥ blend_eps=0.10)

| | vp=margin | vp=relative |
|---|---|---|
| **fp=static**     | **0 (left-censored)** | **0 (left-censored)** |
| **fp=live-delta** | **4** | **6** |

Both `left_censored: true` in the static-floor verdicts; both `left_censored: false` in the live-Δ verdicts. The retrieval-side anchor matters; the write-side anchor (margin vs relative) only widens the live-Δ onset by +2 epochs.

### Supporting trajectory (off-class vote mass, frac_blended)

| run | e0 ocw | e0 fb | e5 ocw | e5 fb | e29 ocw | e29 fb | final occ | final acc | final merge_rate |
|---|---|---|---|---|---|---|---|---|---|
| vp-margin × fp-static     | 0.154 | 0.336 | 0.187 | 0.391 | 0.675 | 0.977 | 377 | 0.375 | 0.836 |
| vp-margin × fp-live-delta | 0.088 | 0.144 | 0.108 | 0.164 | 0.654 | 0.907 | 377 | 0.375 | 0.836 |
| vp-relative × fp-static     | 0.154 | 0.336 | 0.175 | 0.375 | 0.676 | 0.977 | 231 | 0.414 | 0.844 |
| vp-relative × fp-live-delta | 0.088 | 0.144 | 0.086 | 0.131 | 0.680 | 0.976 | 231 | 0.414 | 0.844 |

(ocw = `offclass_weight`, fb = `frac_blended`; final cols at epoch 29.)

Observations:
- **e0 off-class vote mass drops 0.154 → 0.088** with the live-Δ floor — directly mechanistic: the live floor masks low-margin retrieval candidates at the resting manifold geometry, exactly the lever the write gate could not pull.
- **Write-side metrics are invariant under retrieval-floor choice**: occupancy (377/377, 231/231), accuracy (0.375/0.375, 0.414/0.414), merge_rate (identical to 3 d.p.) all reproduce identically across `fp=static`/`live-delta` within each `vp`. The retrieval floor is orthogonal to the write path, as designed.
- **The two compose**: `vp=relative × fp=live-delta` yields the latest onset (e6) AND the lowest e5 off-class weight (0.086), AND the −39% occupancy / +0.039 accuracy from #73's `RelativeVigilance` win.
- **Final-epoch ocw≈0.65 / fb≈0.91**: the live-Δ floor de-censors onset, but does NOT prevent eventual blend saturation under sustained contraction — both runs reach >90% blended by epoch 29. Resolves the central #80 question (onset), leaves open whether the late-epoch trajectory needs additional mechanism (one of the other candidates: confidence-weighted retrieval, manifold-relative temperature, manifold-relative chimera criterion).

### Verdict
The central #80 question — *does the existing retrieval-side mechanism move blend onset off epoch 0?* — is **answered yes** for the live-Δ floor at default `k_logit_steps=3.0`. No need to implement confidence-weighted retrieval or manifold-relative temperature *for the de-censoring goal*. The remaining late-epoch saturation under sustained drift is a separable question; it should be scoped as its own issue if pursued.

<sub>seed 0, CPU, torch 2.8.0 / sentence-transformers 5.1.0. Raw CSVs/verdicts available in the workflow artifact directory.</sub>

**Comment by samscarrow** [comment]
Closing — central question answered yes by the live-Δ `RetrievalFloorPolicy` (#74) at default `k_logit_steps=3.0` on real-20NG + all-MiniLM-L6-v2:

| | vp=margin | vp=relative |
|---|---|---|
| fp=static     | onset e0 (left-censored) | onset e0 (left-censored) |
| fp=live-delta | onset **e4** | onset **e6** |

Full result and supporting trajectory in the [previous comment](https://github.com/git-scarrow/fast-associative-memory/issues/80#issuecomment-4578603709). No need to implement confidence-weighted retrieval, manifold-relative temperature, or manifold-relative chimera criterion for the de-censoring goal.

The separable residual — late-epoch blend saturation under sustained contraction (>90% blended by e29 in both live-Δ runs) — is being tracked in its own issue.
